src/env_node.ml

```diff
@@ -119,45 +116,30 @@ let rec c_flags t ~profile ~expander ~default_context_flags =
   | None ->
     let default =
       match t.inherit_from with
-      | None -> Build.return default_context_flags
+      | None -> C.Kind.Dict.make_both (lazy (Build.return default_context_flags))
       | Some (lazy t) -> c_flags t ~profile ~expander ~default_context_flags
     in
     let flags =
       match find_config t ~profile with
       | None -> default
       | Some cfg ->
-        let expander = Expander.set_dir expander ~dir:t.dir in
-        let eval = Expander.expand_and_eval_set expander in
-        let f = cfg.c_flags in
-        if  Ordered_set_lang.Unexpanded.has_special_forms f then
-          eval f ~standard:default
-        else
-          eval f ~standard:(Build.return [])
+        C.Kind.Dict.mapi cfg.c_flags ~f:(fun ~kind f -> lazy (
+          let expander = Expander.set_dir expander ~dir:t.dir in
```

Comment:
`expander` and `eval` are the same in both case, they can go outside of the closure. BTW, given the previous discussions we had about lazy values, I feel a bit wary of this `lazy`. In practice, it buys us very little, so I'm in favour of making things eager here.

Reply:
I got rid of it. Agreed that it buys very little.